File: bt/bt33.c
//10-09-26
//Code by Anvidha

#include <stdio.h>
#include <math.h>

// Function to calculate the next iteration (x_n+1) using Newton-Raphson
double next_iteration(double x_n) {
    return x_n - 1.0 + 2.0 * exp(-x_n);
}

int main() {
    double x0 = 1.0;// given in question
    double x1 = next_iteration(x0);

    printf("Initial guess (x0): %.6f\n", x0);
    printf("Next iteration (x1): %.6f\n", x1);

    return 0;
}

File: bt/bt33.py
#10-09-26
#Code by Anvidha

import numpy as np
import matplotlib.pyplot as plt

x = np.linspace(0, 1.5, 500)
y = np.exp(x) - 2    #given equation to plot

plt.plot(x, y, label=r'$e^x-2$')
plt.axhline(0, linewidth=1)
plt.axvline(np.log(2), linestyle='--', label=r'Root = $\ln(2)$')

#giving x-axis, y-axis names
plt.xlabel('x')
plt.ylabel('f(x)')
plt.title('Newton-Raphson: $e^x - 2 = 0$')
plt.grid()
plt.legend()

plt.savefig('bt33.pdf', dpi=300, bbox_inches='tight')
